### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by nikhita**
Ref https://github.com/kubernetes/steering/issues/28#issuecomment-416826759, https://github.com/kubernetes/community/issues/1832

The content is as per https://github.com/kubernetes/kubernetes-template-project.

/assign tengqm steveperry-53

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+5. Run `make api` again after fixing the errors
+
+6. Create a new directory under kubernetes/kubernetes.github.io and
+   update kubernetes/kubernetes.github.io
+  - From kubernetes/kubernetes.io `mkdir docs/api-reference/v1.<Minor>`
+  - Update `docs/reference/index.md` to include the new version
+  - Grep for references to the old version, and update them
+
+6. Run `map copyapi` to copy the files to the kubernetes/kubernetes.github.io directory
```
Comment (tengqm) [inline CONTRIBUTING.md]: the target directory should be kuberentes/website

Comment (nikhita) [reply CONTRIBUTING.md]: Fixed :+1:

**Comment by steveperry-53** [comment]
@nikhita Thank you for doing this. I'd like to suggest that we don't give detailed instructions generating docs in contributing.md. Instead, I think we should link to the detailed instructions [here](https://kubernetes.io/docs/contribute/generate-ref-docs/kubernetes-api/) and [here](https://kubernetes.io/docs/contribute/generate-ref-docs/kubectl/). If we publish detailed instructions in two places, we run the risk of the instructions getting out of sync. @tengqm What do you think?

**Comment by tengqm** [comment]
@steveperry-53 I agree.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Welcome to Kubernetes. We are excited about the prospect of you joining our [community](https://github.com/kubernetes/community)! The Kubernetes community abides by the CNCF [code of conduct](code-of-conduct.md). Here is an excerpt:
+
+_As contributors and maintainers of this project, and in the interest of fostering an open and welcoming community, we pledge to respect all people who contribute through reporting issues, posting feature requests, updating documentation, submitting pull requests or patches, and other activities._
+
+## Getting Started
+
+### Generating docs
+
+Please see the following instructions on how to generate reference documentation:
```
Comment (nikhita) [inline CONTRIBUTING.md]: @steveperry-53 @tengqm Added this. Ptal, thanks!

**Comment by tengqm** [comment]
lgtm now.
When kuberentes/website#10123 is in, we may need to tune the guide for generating reference docs for kube-components.

**Pull request merged**